edc-tests/e2e/end2end-transfer-cloud/src/test/java/org/eclipse/tractusx/edc/tests/transfer/S3ToS3EndToEndTest.java

```diff
@@ -118,25 +117,73 @@ void transferFile_success() {
                         .build()
                 ).build();
 
-        var transferProcessId = consumer()
-                .requestAssetFrom(assetId, provider())
+        var transferProcessId = CONSUMER
+                .requestAssetFrom(assetId, PROVIDER)
                 .withTransferType("AmazonS3-PUSH")
                 .withDestination(destination)
                 .execute();
 
         await().atMost(ASYNC_TIMEOUT).untilAsserted(() -> {
-            var state = consumer().getTransferProcessState(transferProcessId);
+            var state = CONSUMER.getTransferProcessState(transferProcessId);
             assertThat(state).isEqualTo(COMPLETED.name());
             assertThat(CONSUMER_CONTAINER.listObjects(destinationBucketName)).isNotEmpty();
         });
     }
 
-    public TractusxParticipantBase provider() {
-        return PROVIDER;
-    }
+    @Test
+    void transferFile_withBucketCreation_success() {
+        var assetId = "s3-test-asset";
+        var sourceBucketNameFuture = PROVIDER_CONTAINER.createBucket();
+        var sourceBucketName = sourceBucketNameFuture;
+
+        PROVIDER_CONTAINER.uploadObjectOnBucket(
+                        sourceBucketName,
+                        TESTFILE_NAME,
+                        TestUtils.getFileFromResourceName(TESTFILE_NAME).toPath());
+
+        var destinationBucketName = "destination-bucket";
+
+        Map<String, Object> dataAddress = Map.of(
+                "name", "transfer-test",
+                "@type", "DataAddress",
+                "type", "AmazonS3",
+                "objectName", TESTFILE_NAME,
+                "region", PROVIDER_CONTAINER.getS3region(),
+                "bucketName", sourceBucketName,
+                "accessKeyId", PROVIDER_CONTAINER.getCredentials().accessKeyId(),
+                "secretAccessKey", PROVIDER_CONTAINER.getCredentials().secretAccessKey(),
+                "endpointOverride", PROVIDER_CONTAINER.getEndpointOverride()
+        );
+
+        PROVIDER.createAsset(assetId, Map.of(), dataAddress);
+        var policyId = PROVIDER.createPolicyDefinition(bnpPolicy(CONSUMER.getBpn()));
```

Comment:
```suggestion
        var policyId = PROVIDER.createPolicyDefinition(bpnPolicy(CONSUMER.getBpn()));
```

Please change other occurrences too

Reply:
Changed every reference in [here](https://github.com/eclipse-tractusx/tractusx-edc/pull/2020/commits/0e191148c0a8bcde1095a4597e618d65b8b23d22).